pkg/streamingpromql/functions/instant_vector.go

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package functions
+
+import (
+	"math"
+
+	"github.com/prometheus/prometheus/promql"
+
+	"github.com/grafana/mimir/pkg/streamingpromql/pooling"
+	"github.com/grafana/mimir/pkg/streamingpromql/types"
+)
+
+type SeriesMetadataFunction func(seriesMetadata []types.SeriesMetadata, pool *pooling.LimitingPool) ([]types.SeriesMetadata, error)
+
+func DropSeriesName(seriesMetadata []types.SeriesMetadata, _ *pooling.LimitingPool) ([]types.SeriesMetadata, error) {
+	for i := range seriesMetadata {
+		seriesMetadata[i].Labels = seriesMetadata[i].Labels.DropMetricName()
+	}
+
+	return seriesMetadata, nil
+}
+
+type InstantVectorFunction func(seriesData types.InstantVectorSeriesData, pool *pooling.LimitingPool) (types.InstantVectorSeriesData, error)
+
+func TransformationFunc(transform func(f float64) float64) InstantVectorFunction {
+	return func(seriesData types.InstantVectorSeriesData, pool *pooling.LimitingPool) (types.InstantVectorSeriesData, error) {
+		for i := range seriesData.Floats {
+			seriesData.Floats[i].F = transform(seriesData.Floats[i].F)
+		}
+
+		return seriesData, nil
+	}
+}
+
+var Acos = TransformationFunc(math.Acos)
+
+func HistogramCount(seriesData types.InstantVectorSeriesData, pool *pooling.LimitingPool) (types.InstantVectorSeriesData, error) {
+	floats, err := pool.GetFPointSlice(len(seriesData.Histograms))
+	if err != nil {
+		return types.InstantVectorSeriesData{}, err
+	}
+
+	data := types.InstantVectorSeriesData{
+		Floats: floats,
+	}
+	for _, Histogram := range seriesData.Histograms {
```

Comment:
[nit] or `h`

```suggestion
	for _, histogram := range seriesData.Histograms {
```

(same applies to `HistogramSum` below)